seps/sep-0002.md

```diff
@@ -0,0 +1,100 @@
+# Support multiple build-systems per package using a directive
+
+## The problem
+
+There are multiple packages that are either changing their build-system during the evolution of the project, or using different build-systems for different platforms. Spack, at the moment, provides no support to model this use case. As a result we wrote some [documentation](https://github.com/spack/spack/pull/25174) to enumerate various workarounds that people adopted - each with its own drawbacks. What we would like to have in the long term is proper support for packages that can be built using multiple build-systems.
```

Comment:
> using different build-systems for different platforms

This is a very good point that I didn't think about while writing https://github.com/spack/spack/pull/25174, glad to see more core developers thinking about Windows support!

Reply:
Very much so! Consider https://github.com/spack/spack/commit/5818a545280f30df1501b5dffecc2d4693beb7fe, are we going to turn all `./configure`-based packages into CMake packages across all platforms just to support Visual Studio? I really hope not... `zlib` is a teeny tiny package that should be able to build with 0 other deps than a shell and a compiler.